src/contracts/mod.rs

```diff
@@ -1,228 +1,112 @@
-    ) -> impl Stream<Item = Result<Log<MemberAddedEvent>, EventError>> + '_ {
-        // Start MemberAdded log event stream
-        let mut filter = self
-            .semaphore
-            .member_added_filter()
-            .from_block(starting_block);
-
-        if let Some(end_block) = end_block {
-            filter = filter.to_block(end_block);
-        }
-        self.ethereum
-            .fetch_events::<MemberAddedEvent>(&filter.filter)
-            .try_filter(|event| future::ready(event.event.group_id == self.group_id))
-    }
-
-    #[instrument(level = "debug", skip_all)]
-    #[allow(dead_code)]
-    pub async fn is_manager(&self) -> AnyhowResult<bool> {
-        info!(address = ?self.ethereum.address(), "My address");
-        let manager = self.semaphore.manager().call().await?;
-        info!(?manager, "Fetched manager address");
-        Ok(manager == self.ethereum.address())
-    }
-
-    #[instrument(level = "debug", skip_all)]
-    pub async fn insert_identity(&self, commitment: Field) -> Result<TransactionReceipt, TxError> {
-        info!(%commitment, "Inserting identity in contract");
-
-        // Send create tx
-        let commitment = U256::from(commitment.to_be_bytes());
-        let receipt = self
-            .sitter
-            .send(self.semaphore.add_member(self.group_id, commitment).tx)
-            .await?;
-        Ok(receipt)
-    }
-
-    // TODO: Ideally we'd have a `get_root` function, but the contract doesn't
-    // support this.
-    #[instrument(level = "debug", skip_all)]
-    pub async fn assert_valid_root(&self, root: Field) -> AnyhowResult<()> {
-        // HACK: Abuse the `verifyProof` function.
-
-        let result = self
-            .semaphore
-            .verify_proof(
-                root.to_be_bytes().into(),
-                self.group_id,
-                U256::zero(),
-                U256::zero(),
-                U256::zero(),
-                [U256::zero(); 8],
-            )
-            .call()
-            .await
-            .expect_err("Proof is invalid");
-        // Result will be either `0x09bde339`: `InvalidProof()` (good) or
-        // `0x504570e3`: `InvalidRoot()` (bad).
-        // See <https://github.com/worldcoin/world-id-example-airdrop/blob/03de53d2cb016ddef28b26e8237e85b62ec385c7/src/Semaphore.sol#L141>
-        // See <https://sig.eth.samczsun.com/>
-        // HACK: There's really no good way to parse these errors
-        let error = result.to_string();
-        if error.contains("0x09bde339") {
-            return Ok(());
-        }
-        if error.contains("0x504570e3") {
-            return Err(anyhow!("Invalid root"));
-        }
-        Err(anyhow!("Error verifiying root: {}", result))
-    }
+    ) -> Option<Pin<Box<dyn Stream<Item = Result<Log<MemberAddedEvent>, EventError>> + Send + '_>>>;
 }
+
+/// A type for an identity manager object that can be sent across threads.
+pub type ThreadableIdentityManager = dyn IdentityManager + Send + Sync;
```

Comment:
This name makes sense, but trips me a little since we never explicitly create threads.

Maybe change to `SharedIdentityManager = Arc<dyn IdentityManager + Send + Sync>` to match `SharedTreeState`? That one reference in the `IdentityCommitter` could be replaced with `Arc`.